ai_powered_qa/components/agent_store.py

```diff
@@ -107,3 +107,36 @@ def save_interaction(self, agent: Agent, interaction: Interaction):
 
         with open(file_path, "w") as file:
             file.write(interaction.model_dump_json(indent=4))
+
+    def load_gherkin_memory(self, agent: Agent):
```

Comment:
This I could live with, although it also seems quite specific to a Gherkin agent, rather then any other agent. I would like to have a clean line between what's framework code, and what's custom code for a specific experiment. This could also be harcoded in the UI script (or a custom script related to gherkin agent), or we could think of a way to generalize Agent saving some other custom data than just history (this would be the better solution long term, but it's probably not the right investment of our time right now)